system/doc/reference_manual/expressions.xml

```diff
@@ -2072,9 +2078,50 @@ KeyPattern := ValuePattern &lt;- MapExpression</pre>
 2> <input>[x || is_integer(x)].</input>
 []</pre>
 
-    <p>More examples are provided in
-    <seeguide marker="system/programming_examples:list_comprehensions">
-    Programming Examples.</seeguide></p>
+    <p>The behavior when a filter expression fails to evaluate to a
+    boolean value depends on whether the expression is a <seeguide
+    marker="#guard_expressions">guard expression</seeguide> or
+    not:</p>
```

Comment:
The sentence is correct, but it seems like it could be improved. 

I propose the suggestion below, but feel free to either leave as is or change to something that you like better 👍🏽 :
```html
<p>Filter expressions may change the behaviour of their evaluation logic 
when their result is a non-boolean value as follows:</p>
```

Reply:
Yes, the sentence felt awkward to me. Changed to:

> What happens when the filter expression does not evaluate to a boolean value depends on the expression: